Mathlib/Algebra/Algebra/Subalgebra/Rank.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
+
+/-!
+
+# Some results on the ranks of subalgebras
+
+This file contains some results on the ranks of subalgebras,
+which are corollaries of `rank_mul_rank`.
+Since their proof essentially depends on the fact that a non-trivial commutative ring
+satisfies strong rank condition, we put them into a separate file.
+
+-/
+
+open FiniteDimensional
+
+namespace Subalgebra
+
+variable {R S : Type*} [CommRing R] [CommRing S] [Algebra R S]
+
+variable (A B : Subalgebra R S)
+
+theorem rank_sup_eq_rank_left_mul_rank_of_free
+    [Module.Free R A] [Module.Free A (Algebra.adjoin A (B : Set S))] :
```

Comment:
Can you put some (all?) of this assumptions in a `variable` line?

Reply:
Let me try. I'm afraid that it will pick up wrong assumptions.